### PR TITLE
Update deps

### PR DESCRIPTION
**Pull request by divmain**
Only one of f9d041a or 8fe6745 are necessary - either one will fix the broken tests.  My preference would be the latter, but I'll leave that up to you.  Let me know which one you prefer and I'll squash before this is merged.

Although this turned out to be a simple fix after a long rabbit hole, it was a beneficial introduction to circus and how you're going about things!  I have some questions and some feedback, but I'll send that separately.

@kpdecker

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -98,10 +98,9 @@ describe('loader integration', function() {
           expect(loaded.scripts[3]).to.match(/require.js$/);
           expect(loaded.scripts[4]).to.match(/exec.js$/);
 
-          expect(loaded.log).to.eql([
-            '_: true Handlebars: true',
-            'App: _: true Handlebars: true Vendor: true'
-          ]);
+          expect(loaded.log)
+            .to.contain('_: true Handlebars: true').and
+            .to.contain('App: _: true Handlebars: true Vendor: true');
```
Comment (kpdecker) [inline test/index.js]: Does the order of the execution change?

Comment (divmain) [inline test/index.js]: If you swap the array elements, the tests do pass consistently.  However, that behavior is extrinsic: two async callbacks are placed on the event loop and, at present, the "App" one fires first.  But a seemingly unrelated change down the line could alter the behavior, and the test would mysteriously start failing.  The new assertions seemed a better reflection of what is actually supposed to happen.

Comment (kpdecker) [inline test/index.js]: It's very important that the load order here remains the same. `amd-exec` is explicitly dependent on `vendor` meaning we need `vendor` to execute before `amd-exec` does and then the application will be in an undetermined state. It sounds like we need to fix the loading behavior so it's deterministic.

I've been tempted to alter the tests in the past as well but each time convince myself it's the wrong route. Perhaps we add a comment to the test saying that the order is important.

File: test/fixtures/amd-exec.js [hunk 2]
```diff
@@ -3,13 +3,19 @@ require.config({
...
     'chunk_vendor1': 'bootstrap'
   }
 });
 
-require(['lodash', 'handlebars/runtime', 'vendor'], function(_, Handlebars, Vendor) {
+require([
+  'lodash',
+  'handlebars/runtime',
+  'vendor',
+  'vendor/test/fixtures/packages'
```
Comment (kpdecker) [inline test/fixtures/amd-exec.js]: Tying this into the other comments, does adding this dependency force the proper loading order consistently but it does not with just the module reference?

It should work without this change since we shouldn't expect users to have to deal with this loading behavior.

**Comment by divmain** [comment]
Responding to both comments here @kpdecker.

Originally in the AMD module, we do the following (leaving out lodash and hbs for simplicity):
```javascript
require(["vendor"], function () { ...
  // AMD module
});
```

That `vendor` reference is specifically to the following code:
```javascript
require.ensure('./packages', function() {
  require('./packages');
});
```

The circus-amd plugin will guarantee that the code above will execute before the AMD module executes.  However, there is no guarantee that the inner function (`function() { require('./packages'); }`) will execute before the AMD module does.  `require.ensure` ensures that `./packages` is loaded before the inner function executes, but does not ensure that `./packages` is loaded before the outer webpack module (`vendor`) returns its exported value.

In other words, `require.ensure` guarantees that, once the inner function fires, `require('./packages');` will be synchronous.  It does not, however, guarantee that the inner function itself fires synchronously in relation its containing module.  The inner function will be fired asynchronously, as will the AMD module.

**Comment by kpdecker** [comment]
OH! Ok, I missed that there was an embedded `require.ensure` in there. In that case, I'm fine with either change. If we do the contains assertion change then we should have a test case explaining that there isa `require.ensure` in the test fixture thus the loading order is undefined.

We might want to consider adding (if there is not already) a test that verifies the order behavior with just the `vendor` require and no `require.ensure`, if one does not exist.

**Comment by divmain** [comment]
Cool - I added a test to assert that the root webpack dependency loads first, and then both of them (in either order) also load.  Also added comments to the tests to explain what should happen first and why we're testing for only presence of the other two log messages instead of ordering.

Merging, but let me know if you'd like me to come back and add additional comments somewhere.
